packages/flutter/lib/src/material/icon_theme.dart

```diff
@@ -22,13 +25,33 @@ class IconTheme extends InheritedWidget {
     assert(child != null);
   }
 
+  /// Creates an icon theme that controls the color, opacity, and size of
+  /// descendant widgets, and merges in the current icon theme, if any.
+  ///
+  /// The [context], [data], and [child] arguments must not be null.
+  factory IconTheme.merge({
+    Key key,
+    @required BuildContext context,
+    @required IconThemeData data,
+    Widget child
```

Comment:
We usually make `child` arguments to `InheritedWidgets` `@required`.  There's not much point in making an inherited widget that doesn't have anyone to inherit from it.